api/v1beta1/azuremanagedcontrolplane_types.go

```diff
@@ -213,6 +213,10 @@ type AzureManagedControlPlaneSpec struct {
 	// OIDCIssuerProfile is the OIDC issuer profile of the Managed Cluster.
 	// +optional
 	OIDCIssuerProfile *OIDCIssuerProfile `json:"oidcIssuerProfile,omitempty"`
+
+	// DNSPrefix allows the user to customize dns prefix. This cannot be updated once the Managed Cluster has been created.
```

Comment:
not to match formatting with other immutable fields:
```suggestion
	// DNSPrefix allows the user to customize dns prefix.
	// Immutable.
```